### DISCUSSION TITLE
Question on assigning orientation on grains from experiment in multiscale model

### DISCUSSION THREAD

**Discussion by Samelail** [neperfepx/neper#805] [Q&A] [answered]
Hi,

I have created a multiscale 2D microstructure with seeds and size controlled. However, I also need to input the grain orientation from IPF. It is not a full field analysis, therefore, no pixel by pixel orientation information is required despite the IPF comes from the EBSD. In the output file, the angle under '*ori' is now random distributed but the number is exactly the same as my final seed number. 

![img3](https://github.com/neperfepx/neper/assets/60886727/57a028be-bdd6-4773-b93c-69028899192f)
![Coarse IPF](https://github.com/neperfepx/neper/assets/60886727/1a5daa77-f0bf-4b9c-8817-1ba14d139676)

Code used is shown as following:
neper -T -n 66::from_morpho -morphooptiini "coo:file(coo2),weight:file(weights)" -morpho "gg::msfile(morpho2)" -dim 2 -domain "square(40,40)" -reg 1 -morphooptistop "eps<1e-3" -o mtp3

**Comment by rquey** [comment, marked as answer]
Sorry for the delay...

You can use `-ori odf`. However, the texture needs to be defined as an ODF, not from an IPF.

**Reply by Samelail** [reply]
Hi,
Thanks for the reply. 
I have tested with -ori odf using some random distribution. But I cannot comprehensively understand the meaning for the cmd odf(mesh=file(**.msh),val=file(odf.txt)). 
From the tutorial example, msh is actually a mesh file with orientation but only one line as:
 $ElsetOrientations
1 rodrigues:active
1    0.919953402851    1.412193625247   -0.358468593221
$EndElsetOrientations
odf is a 1D array file with 44315 lines.

When I get the ODF from experiments, it is formated like:
% phi1    Phi     phi2    value
0.00000 0.00000 0.00000 3.46659
5.00000 0.00000 0.00000 3.20484
10.00000 0.00000 0.00000 2.63068
...

It seems that we need an mesh file created somewhere else and use it here?
How to format the odf to the one that we use in Neper? The odf.txt here is somehow different from what normally we get from experiments.
Furthermore, can we input the ODF with seeds? I have different ODFs for coarse grain fine grain from the experiments and currently I am generating the microsturcture through controling seed position and morpho.

Best regards.

**Comment by anujyadav0693** [comment]
Hi [Samelail]
Can you share the Neper code for multiscale 2D microstructure with seeds and size controlled?
